src/main/java/io/cryostat/agent/MainModule.java

```diff
@@ -186,6 +276,43 @@ public static HttpClient provideHttpClient(
         return builder.build();
     }
 
+    @Provides
+    @Singleton
+    public static HttpServer provideHttpServer(
+            ScheduledExecutorService executor,
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_HOST) String host,
+            @Named(ConfigModule.CRYOSTAT_AGENT_WEBSERVER_PORT) int port,
+            @Named(HTTP_SERVER_SSL_CTX) Optional<SSLContext> sslContext) {
+        try {
+            HttpServer http;
+            if (sslContext.isEmpty()) {
+                http = HttpServer.create(new InetSocketAddress(host, port), 0);
+            } else {
+                HttpsServer https = HttpsServer.create(new InetSocketAddress(host, port), 0);
+                https.setHttpsConfigurator(
+                        new HttpsConfigurator(sslContext.get()) {
+                            public void configure(HttpsParameters params) {
+                                try {
+                                    SSLContext context = getSSLContext();
+                                    SSLEngine engine = context.createSSLEngine();
+                                    params.setNeedClientAuth(false);
+                                    params.setCipherSuites(engine.getEnabledCipherSuites());
+                                    params.setProtocols((engine.getEnabledProtocols()));
+                                    params.setSSLParameters(context.getDefaultSSLParameters());
+                                } catch (Exception e) {
+                                    throw new RuntimeException(e);
+                                }
+                            }
+                        });
+                http = https;
```

Comment:
```suggestion
                http = HttpsServer.create(new InetSocketAddress(host, port), 0);
                ((HttpsServer) http).setHttpsConfigurator(
                        new HttpsConfigurator(sslContext.get()) {
                            public void configure(HttpsParameters params) {
                                try {
                                    SSLContext context = getSSLContext();
                                    SSLEngine engine = context.createSSLEngine();
                                    params.setNeedClientAuth(false);
                                    params.setCipherSuites(engine.getEnabledCipherSuites());
                                    params.setProtocols((engine.getEnabledProtocols()));
                                    params.setSSLParameters(context.getDefaultSSLParameters());
                                } catch (Exception e) {
                                    throw new RuntimeException(e);
                                }
                            }
                        });
```

Not sure if I really like this any better after having written it out. Feel free to take it or leave it.